nomad/acl_endpoint.go

```diff
@@ -27,6 +39,19 @@ func (a *ACL) UpsertPolicies(args *structs.ACLPolicyUpsertRequest, reply *struct
 		return fmt.Errorf("must specify as least one policy")
 	}
 
+	// Validate each policy
+	for idx, policy := range args.Policies {
+		if !validPolicyName.MatchString(policy.Name) {
```

Comment:
@armon Would have preferred a `Validate() error` method on the policy and using multi error